### PR TITLE
Snap Grid for Zimberlab Blocks

### PR DESCRIPTION
**Pull request by fayeli**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: d1c33ab399a7ccf30b39ec1b8415e435fe7c111f available at: https://gist.github.com/0e6e817e41ee8fe272dcc585f45acff0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8919-d1c33ab.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8919-d1c33ab.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f9f9f4027eb2204b0b887d0606d697ff2b78b174 available at: https://gist.github.com/0e6e817e41ee8fe272dcc585f45acff0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8919-f9f9f40.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8919-f9f9f40.dmg

**Comment by fayeli** [comment]
https://github.com/fayeli/hifi_scripts/blob/master/snapBlockMarketplaceScript.js
https://github.com/fayeli/hifi_scripts/blob/master/setGridEntityScript.js

Links to other scripts (not in system) for Zimberlab Snap Block project

**Comment by hifi-gustavo** [comment]
Build console output for commit: f9f9f4027eb2204b0b887d0606d697ff2b78b174 available at: https://gist.github.com/0e6e817e41ee8fe272dcc585f45acff0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8919-f9f9f40.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8919-f9f9f40.dmg

**Comment by AndrewMeadows** [comment]
The failed ubuntu build can be ignored -- this PR only changes one JS file.

File: scripts/system/libraries/gridTool.js [hunk 1]
```diff
@@ -205,8 +207,31 @@ Grid = function(opts) {
...
+            print("grid set snap to grid");
+            that.setMajorIncrement(object.majorGrid);
+            that.setMinorIncrement(object.minorGrid);
+            print("grid set major size: " + object.majorGrid + " minor size: " + object.minorGrid);
+        }
+    }
+
+    function listenForSnapBlock() {
+        snapBlockChannel = "Snap-Block-Channel";
+        print('grid subsribe to ' + snapBlockChannel);
```
Comment (AndrewMeadows) [inline scripts/system/libraries/gridTool.js]: subsribe --> subscribe

**Comment by hifi-gustavo** [comment]
Build console output for commit: ee7030d90a916544246518a6d77cfd9ece6a082b available at: https://gist.github.com/0e6e817e41ee8fe272dcc585f45acff0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8919-ee7030d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8919-ee7030d.dmg

File: scripts/system/libraries/gridTool.js [hunk 2]
```diff
@@ -205,8 +207,33 @@ Grid = function(opts) {
         }
     }
 
+    function handleMessages(channel, message, sender) {
+        if (channel === snapBlockChannel && sender === MyAvatar.sessionUUID) {
+            print("grid recieved message: " + message);
```
Comment (PhilipRosedale) [inline scripts/system/libraries/gridTool.js]: should these always be printed?  Looks like this could cause log spam?

Comment (fayeli) [inline scripts/system/libraries/gridTool.js]: It should only run once with our snap block use case. I kept it printing just for future debugging, I can remove the line if that's preferred though

**Comment by fayeli** [comment]
Change in project direction

**Pull request closed without merging**